src/ASM/ASMs3Dmx.C

```diff
@@ -271,75 +271,79 @@ bool ASMs3Dmx::generateFEMTopology ()
+        for (int i = spline.order(0) - 1; i >= 0; i--)
+          myMNPC[iel][elm_node++] = last_node - (k*nv + j)*nu - i;
+  };
+
   inod = std::accumulate(nb.begin(),nb.begin()+geoBasis-1,0u);
-  Go::SplineVolume* b = m_basis[geoBasis-1].get();
-  auto knotw = b->basis(2).begin();
-  for (i3 = 1; i3 <= b->numCoefs(2); i3++, ++knotw) {
-    auto knotv = b->basis(1).begin();
-    for (i2 = 1; i2 <= b->numCoefs(1); i2++, ++knotv) {
-      auto knotu = b->basis(0).begin();
-      for (i1 = 1; i1 <= b->numCoefs(0); i1++, inod++, ++knotu)
-        if (i1 >= b->order(0) && i2 >= b->order(1) && i3 >= b->order(2))
-        {
-          if (b->knotSpan(0,i1-1) > 0.0)
-            if (b->knotSpan(1,i2-1) > 0.0)
-              if (b->knotSpan(2,i3-1) > 0.0)
-              {
-                myMLGE[iel] = ++gEl; // global element number over all patches
-
-                int lnod = firstItgElmNode;
-                myMNPC[iel].resize(nElmNode,0);
-                for (j3 = b->order(2)-1; j3 >= 0; j3--)
-                  for (j2 = b->order(1)-1; j2 >= 0; j2--)
-                    for (j1 = b->order(0)-1; j1 >= 0; j1--)
-                      myMNPC[iel][lnod++] = inod - b->numCoefs(0)*b->numCoefs(1)*j3 - b->numCoefs(0)*j2 - j1;
-                // find knotspan spanning element for other bases
-                lnod = 0;
-                size_t basis_ofs = 0;
-                for (size_t bas = 0; bas < m_basis.size(); ++bas) {
-                  if (bas != static_cast<size_t>(geoBasis-1)) {
-                    double ku = *knotu;
-                    double kv = *knotv;
-                    double kw = *knotw;
-                    int bknotu = m_basis[bas]->basis(0).knotIntervalFuzzy(ku);
-                    int bknotv = m_basis[bas]->basis(1).knotIntervalFuzzy(kv);
-                    int bknotw = m_basis[bas]->basis(2).knotIntervalFuzzy(kw);
-                    size_t iinod = basis_ofs + (bknotw*m_basis[bas]->numCoefs(1)+bknotv)*m_basis[bas]->numCoefs(0) + bknotu;
-                    for (j3 = m_basis[bas]->order(2)-1; j3 >= 0; j3--)
-                      for (j2 = m_basis[bas]->order(1)-1; j2 >= 0; j2--)
-                        for (j1 = m_basis[bas]->order(0)-1; j1 >= 0; j1--)
-                          myMNPC[iel][lnod++] = iinod - (j3*m_basis[bas]->numCoefs(1)+j2)*m_basis[bas]->numCoefs(0) - j1;
-                  } else
-                    lnod += elem_size[bas];
-                  basis_ofs += nb[bas];
-                }
+  auto knotw = itg->basis(2).begin();
+  for (int k = 1; k <= itg->numCoefs(2); k++, ++knotw) {
+    auto knotv = itg->basis(1).begin();
+    for (int j = 1; j <= itg->numCoefs(1); j++, ++knotv) {
+      auto knotu = itg->basis(0).begin();
+      for (int i = 1; i <= itg->numCoefs(0); i++, inod++, ++knotu)
+        if (i >= itg->order(0) && j >= itg->order(1) && k >= itg->order(2)) {
+          if (itg->knotSpan(0,i-1) > 0.0 &&
+              itg->knotSpan(1,j-1) > 0.0 &&
+              itg->knotSpan(2,k-1) > 0.0) {
```

Comment:
Similar comment as in the 2D case, consider revising the if-statements

Reply:
similar answer as well. sorry no can do.